File: Move All 0 to the End/move_all_0_to_end.c
#include <stdio.h>
int main() {
    int n;
    printf("Enter the number of elements: ");
    scanf("%d", &n);
    int a[n], result[n];
    for (int i = 0; i < n; i++) {
        printf("Enter element %d: ", i + 1);
        scanf("%d", &a[i]);
    }
    int index = 0;
    int zero_count = 0;
    for (int i = 0; i < n; i++) {
        if (a[i] == 0) {
            zero_count++;
        } else {
            result[index] = a[i];
            index++;
        }
    }
    for (int i = 0; i < zero_count; i++) {
        result[index] = 0;
        index++;
    }
    printf("Array after moving all 0 to the end:\n");
    for (int i = 0; i < n; i++) {
        printf("%d\n", result[i]);
    }
    return 0;
}

File: Move All 0 to the End/MoveAll0ToEnd.java
public class MoveAll0ToEnd {
    public static void main(String[] args) {
        java.util.Scanner sc = new java.util.Scanner(System.in);
        System.out.print("Enter the number of elements: ");
        int n = sc.nextInt();
        int[] a = new int[n];
        int[] result = new int[n];
        for (int i = 0; i < n; i++) {
            System.out.print("Enter element " + (i + 1) + ": ");
            a[i] = sc.nextInt();
        }
        int index = 0;
        int zeroCount = 0;
        for (int i = 0; i < n; i++) {
            if (a[i] == 0) {
                zeroCount++;
            } else {
                result[index] = a[i];
                index++;
            }
        }
        for (int i = 0; i < zeroCount; i++) {
            result[index] = 0;
            index++;
        }
        System.out.println("Array after moving all 0 to the end:");
        for (int i = 0; i < n; i++) {
            System.out.println(result[i]);
        }
    }
}
